assets/js/frontend/single-product.js

```diff
@@ -65,41 +65,59 @@ jQuery( function( $ ) {
 
 				return false;
 			}
-		})
-		.on( 'woocommerce_init_gallery', function() {
-			if ( $.isFunction( $.fn.zoom ) && wc_single_product_params.zoom_enabled ) {
-				wc_product_gallery.init_zoom();
-			}
-		});
+		} );
 
 	//Init Tabs and Star Ratings
 	$( '.wc-tabs-wrapper, .woocommerce-tabs, #rating' ).trigger( 'init' );
 
 	/**
 	 * Product gallery class.
 	 */
-	var wc_product_gallery = {
+	var Product_Gallery = function( $el, args ) {
+
+		var gallery = this;
+
+		this.$el     = $el;
+		this.$images = $( '.woocommerce-product-gallery__image', $el );
+
+		// Make this object available.
+		$el.data['product_gallery'] = this;
+
+		// Pick functionality to initialize...
+		this.flexslider_enabled = $.isFunction( $.fn.flexslider ) && wc_single_product_params.flexslider_enabled;
+		this.zoom_enabled       = $.isFunction( $.fn.zoom ) && wc_single_product_params.zoom_enabled;
+		this.photoswipe_enabled = typeof PhotoSwipe !== 'undefined' && wc_single_product_params.photoswipe_enabled;
+
+		// ...also taking args into account.
+		if ( args ) {
+			this.flexslider_enabled = false === args.photoswipe_enabled ? false : this.flexslider_enabled;
+			this.zoom_enabled       = false === args.zoom_enabled ? false : this.zoom_enabled;
+			this.photoswipe_enabled = false === args.photoswipe_enabled ? false : this.photoswipe_enabled;
+		}
 
```

Comment:
Gallery functionality can be added to any jQuery DOM object. Specific gallery features (flexslider, zoom, photoswipe) can be disabled using `args`.